### PR TITLE
Add a new pacakage: HiC-Pro.

### PR DESCRIPTION
**Pull request by weijianwen**

### CODE REVIEW

File: var/spack/repos/builtin/packages/hic-pro/package.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    depends_on('bowtie2')
+    depends_on('samtools')
+    depends_on('python@2.7:2.8')
+    depends_on('py-numpy')
+    depends_on('py-scipy')
+    depends_on('py-pysam')
+    depends_on('py-bx-python')
+    depends_on('r')
+    depends_on('r-rcolorbrewer')
+    depends_on('r-ggplot2')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hic-pro/package.py]: All of these Python/R dependencies should probably be `type=('build', 'run')`.

File: var/spack/repos/builtin/packages/hic-pro/package.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        config = FileFilter('config-system.txt')
+        config.filter('/HiC-Pro_2.10.0', '')
+        # Install
+        install('config-hicpro.txt', prefix)
+        install('config-install.txt', prefix)
+        install('config-system.txt', prefix)
+        install_tree('bin', prefix.bin)
+        install_tree('annotation', join_path(prefix, 'annotation'))
+        install_tree('doc', join_path(prefix, 'doc'))
+        install_tree('scripts', join_path(prefix, 'scripts'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hic-pro/package.py]: You can actually use `prefix.annotation`, `prefix.doc`, and `prefix.scripts` if you want, no need for `join_path`. You'll still need it for `test-op` though.

File: var/spack/repos/builtin/packages/hic-pro/package.py (hunk 1 above)
Comment (weijianwen) [reply var/spack/repos/builtin/packages/hic-pro/package.py]: Fixed.

File: var/spack/repos/builtin/packages/hic-pro/package.py (hunk 2 above)
Comment (weijianwen) [reply var/spack/repos/builtin/packages/hic-pro/package.py]: Fixed.

**Pull request merged**